pyaedt/modules/Material.py

```diff
@@ -232,6 +232,12 @@ def __init__(self, material, name, val=None, thermalmodifier=None):
         elif val is not None and val["property_type"] == "AnisoProperty":
             self.type = "anisotropic"
             self.value = [val["component1"], val["component2"], val["component3"]]
+        elif val is not None and val["property_type"] == "nonlinear":
```

Comment:
```suggestion
        elif val["property_type"] == "nonlinear":
```

Reply:
@MaxJPRey I cannot commit this modification. If val is None then it would fail in looking for "property_type". So I need first to check that val is filled and then search for the type

Reply:
Yes you are write, it would require a try/except block to prevent from  that and raise and exception.

Reply:
try/except is more expensive when it raises compared to type check